### PR TITLE
DOC: Fixed errors in pandas.DataFrame.asfreq PR07, RT02, RT03, SA04

### PR DESCRIPTION
**Pull request by tolhassianipar**
- [X] closes https://github.com/pandanistas/pandanistas_sprint_ui2020/issues/19
- [ ] tests added / passed
- [X] passes `black pandas`
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Output of `python scripts/validate_docstrings.py pandas.DataFrame.asfreq`:
```
################################################################################
##################### Docstring (pandas.DataFrame.asfreq)  #####################
################################################################################

Convert TimeSeries to specified frequency.

Optionally provide filling method to pad/backfill missing values.

Returns the original data conformed to a new index with the specified
frequency. ``resample`` is more appropriate if an operation, such as
summarization, is necessary to represent the data at the new frequency.

Parameters
----------
freq : DateOffset or str
    Frequency DateOffset or string.
method : {'backfill'/'bfill', 'pad'/'ffill'}, default None
    Method to use for filling holes in reindexed Series (note this
    does not fill NaNs that already were present):

    * 'pad' / 'ffill': propagate last valid observation forward to next
      valid
    * 'backfill' / 'bfill': use NEXT valid observation to fill.
how : {'start', 'end'}, default end
    For PeriodIndex only (see PeriodIndex.asfreq).
normalize : bool, default False
    Whether to reset output index to midnight.
fill_value : scalar, optional
    Value to use for missing values, applied during upsampling (note
    this does not fill NaNs that already were present).

Returns
-------
converted
    Same type as caller.

See Also
--------
reindex : Conform DataFrame to new index with optional filling logic.

Notes
-----
To learn more about the frequency strings, please see `this link
<https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#offset-aliases>`__.

Examples
--------
Start by creating a series with 4 one minute timestamps.

>>> index = pd.date_range('1/1/2000', periods=4, freq='T')
>>> series = pd.Series([0.0, None, 2.0, 3.0], index=index)
>>> df = pd.DataFrame({'s':series})
>>> df
                       s
2000-01-01 00:00:00    0.0
2000-01-01 00:01:00    NaN
2000-01-01 00:02:00    2.0
2000-01-01 00:03:00    3.0

Upsample the series into 30 second bins.

>>> df.asfreq(freq='30S')
                       s
2000-01-01 00:00:00    0.0
2000-01-01 00:00:30    NaN
2000-01-01 00:01:00    NaN
2000-01-01 00:01:30    NaN
2000-01-01 00:02:00    2.0
2000-01-01 00:02:30    NaN
2000-01-01 00:03:00    3.0

Upsample again, providing a ``fill value``.

>>> df.asfreq(freq='30S', fill_value=9.0)
                       s
2000-01-01 00:00:00    0.0
2000-01-01 00:00:30    9.0
2000-01-01 00:01:00    NaN
2000-01-01 00:01:30    9.0
2000-01-01 00:02:00    2.0
2000-01-01 00:02:30    9.0
2000-01-01 00:03:00    3.0

Upsample again, providing a ``method``.

>>> df.asfreq(freq='30S', method='bfill')
                       s
2000-01-01 00:00:00    0.0
2000-01-01 00:00:30    NaN
2000-01-01 00:01:00    NaN
2000-01-01 00:01:30    2.0
2000-01-01 00:02:00    2.0
2000-01-01 00:02:30    3.0
2000-01-01 00:03:00    3.0

################################################################################
################################## Validation ##################################
################################################################################
```

### CODE REVIEW

**Comment by WillAyd** [comment]
Can you merge master? I think your changed to pandas._testing were already done in #32358 so not sure why they still show here without conflict (maybe a GitHub bug?)

File: pandas/core/generic.py [hunk 1]
```diff
@@ -7381,6 +7381,7 @@ def asfreq(
         Parameters
         ----------
         freq : DateOffset or str
+            frequency string
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py]: Not only a string, can also be a DateOffset object

(also, can you have it start with a capital + end with a period?)

**Comment by tolhassianipar** [comment]
Hello, sorry for late reply. I will try to fix on weekend because some reasons. Thank you.

File: pandas/core/generic.py [hunk 2]
```diff
@@ -7398,11 +7399,12 @@ def asfreq(
 
         Returns
         -------
-        converted : same type as caller
+        converted
+            Same type as caller.
```
Comment (datapythonista) [inline pandas/core/generic.py]:
```suggestion
        Same type as caller
            Object converted to the specified frequency.
```

**Comment by datapythonista** [comment]
@tolhassianipar there are still unaddressed comments, not sure if you've seen it.

Also, you'll have to merge master again, since the statsmodel url is still failing sometimes, and making the CI red.

**Review by datapythonista — APPROVED** [review]
Thanks @tolhassianipar, lgtm

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @tolhassianipar

**Comment by tolhassianipar** [comment]
Happy to join :) thanks for the guidances

**Pull request merged**